mpi-proxy-split/split_process.cpp

```diff
@@ -71,19 +73,52 @@ static char* proxyAddrInApp();
 static void compileProxy();
 #endif
 
+/* The support to set and get FS base register in user-space has been merged in
+ * Linux kernel v5.3 (see https://elixir.bootlin.com/linux/v5.3/C/ident/rdfsbase
+ *  or https://www.phoronix.com/scan.php?page=news_item&px=Linux-5.3-FSGSBASE).
+ *
+ * MANA leverages this faster user-space switch on kernel version >= 5.3.
+ */
+static inline unsigned long getFS(void)
+{
+  unsigned long fsbase;
+  // The prefix 'rex.W' is required or 'rdfsbase' will assume 32 bits.
+  asm volatile("rex.W\n rdfsbase %0" : "=r" (fsbase) :: "memory");
+  return fsbase;
+}
+
+static inline void setFS(unsigned long fsbase)
+{
+  // The prefix 'rex.W' is required or 'wrfsbase' will assume 32 bits.
+  asm volatile("rex.W\n wrfsbase %0" :: "r" (fsbase) : "memory");
+}
+
 SwitchContext::SwitchContext(unsigned long lowerHalfFs)
 {
   this->lowerHalfFs = lowerHalfFs;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,9,0) || defined(HAS_FSGSBASE)
```

Comment:
@xuyao0127 , I agree that this is "defensive programming", which is not always recommended.
In theory, `defined(HAS_FSGSBASE)` should be sufficient.  But because that has not been heavvily tested, I added "insurance".  If the kernel is at least 5.9.0, then we know that we have user-space FSGSBASE.

Should I add the following short comment?
```
// Linux kernel 5.9 or higher guarantees that FSGSBASE is supported.
// For now, include "|| defined(HAS_FSGSBASE)" to test if FSGSBASE support was backported.
```

Reply:
I think a short comment is helpful.